### ISSUE TITLE
Review Team Agreements + DoD + DoR

### ISSUE THREAD

**Issue by eazeredo** [bcgov/CONN-CCBC-portal#614]
- [ ] Updating team agreements for updating dependencies and runtimes

**Comment by AntBush** [comment]
[Team Agreements](https://github.com/bcgov/CONN-CCBC-portal/blob/main/docs/Team_Agreements.md):
I feel like the Team Agreements have a good base, with some additions and edits that can make it more clear. It provides a good definition of day-to-day tasks (card creation, branch naming policy, etc)
To be added to Team Agreements:
- Now that it's included, the use of the pre-commit hooks
- Estimating/reviewing cards before a review/refinement session
- Tracking work done on an issue using checkboxes
- Standups start with Designer work first
- (needs expanding) some more meeting agreements that we are enacting but haven't written down.
Edits to Team Agreements, anything in parentheses are my thoughts on the change:
- Branch naming policy is as follows: <s>[issue#]-brief-description</s> [fix/feat/ci/chore/etc]/[issue#]-brief-description. <s>In the event that an issue does not already exist for the changes being made in a branch, one should be made.</s> (There are 2 ways that we're naming the branches right now, and I think the fix/feat/ci/etc makes it easier to find a branch, additionally I think having a hard rule for creating issues for *all* branches isn't efficient or beneficial in many cases where a branch is made without an issue)
- Reviews can be done by <s>the first person</s> anyone who gets to it. If the code needs explaining, request the author to walk you through it. (first person implies only one reviewer is needed, we want to move towards multiple reviewers per PR)
- When necessary, use git rebase to sync a feature branch with main. This keeps a tidier history in git, however, can be destructive, so do so cautiously. (This is listed under Development and Version Control, but should be moved to Guardrails with the setup of our repo)
- <s>Communication takes place in the development channel in the Gov Team. Gov Teams is the main channel of communication between PO, Stakeholders and the team </s> (this isn't happening and should either be removed or changed to reflect the behaviour we intend. If this is the intended behaviour, we should work to enforce this better)

[Definition of Ready](https://github.com/bcgov/CONN-CCBC-portal/blob/main/docs/Definition_of_%20Ready.md):
The DOR I find is sufficient for any bug/user story tickets. There are some points I'd like to add:
- Under "For each Individual sprint" I think a card isn't ready for work until it's been estimated/gone through planning porker. This is highlighted under "The Acceptance Criteria was evaluated by the Dev Team" but I don't find it sufficiently clear. 
- The DOR doesn't fit well with a "Discussion" card, where the action isn't development work but rather hearing thoughts from multiple team members. I think a specific section (whether it's here or in the Team Agreements) should be created to establish how a Discussion card should be written. In my opinion a discussion card requires 3 elements: 
1. **Clear Questions to be Answered**: discussion of a general topic can be had but is often too vague to start without knowing the prior intention of the one who created the card, which leads me to
2. **Bootstrap of Conversation**: whether it's the one who wrote the card or a designated team member, an initial answer to the question/topic of discussion should be given to give some direction to the conversation
3. **Clear outcome**: once a conclusion has been reached, there should be some actionable item that follows (report written, update to team agreement, new ticket, branch, etc). This is also to steer the conversation in a meaningful way.

[Definition of Done](https://github.com/bcgov/CONN-CCBC-portal/blob/main/docs/Definition_of_Done.md):
There are some things in the DoD  that can be dropped, as it seems to follow closely what is in our PR template. I'll be doing this one per section: 
- The "What do we need to check for each PR" can be removed as those are either included in our guardrails. If we really want to, we can have that section just point to our PR template. 
- "What do we need to check to move the card to "PO Review" column?" needs to be edited to remove the deploy to test as that is no longer our CD strategy.
- "What do we need to check to deploy in Dev?" Can be removed, as those guardrails are apart of our CI
- "What do we need to check to deploy in Test?" Remove "User Manual/FAQ updated" as well as "Presentation to Stakeholders" (we'll be deploying too often for presentations)
- "What do we need to check to deploy in Prod?" I think the only thing we need here is PO approval
- "Branching Model Team Agreement" Everything below this can be deleted, as it is mostly repeated from our Team Agreements

**Comment by MarsRomer** [comment]
Add to team agreements using the links in the description above:  How to handle security tickets. - make tickets in zen hub in next sprint candidates

**Comment by VB-at-Bis** [comment]
My only comment is about `When necessary, use git rebase to sync a feature branch with main. ` . Back merge from `main` should be acceptable in complex cases to simplify conflict resolution. `git push --force` "is a rather [dangerous process](https://www.git-tower.com/learn/git/faq/git-force-push#:~:text=The%20%2D%2Dforce%20option%20for,lose), because it's very easy to overwrite (and thereby lose) commits"

**Comment by matthieu-foucault** [comment]
@VB-at-Bis (and team) do you have a preference regarding squashing of accepted PRs? If we update PRs with a merge commit, I would prefer merging PRs with a squash to keep the git history readable. 

This would have the added benefits of:
- making our generated changelog better, as long as we have good conventions for our PR titles and description, and use those as the squash commit message. 
- allowing us to have a merge bot that updates our PRs and automerges

**Comment by VB-at-Bis** [comment]
Yes, merge with a squash is definitely preferred way (imho).

**Comment by matthieu-foucault** [comment]
Other team agreements I'd like to cover:
 - PR review: what should they cover? As a result of our unsustainable pace, I think we've had too many "LGTM" reviews where we don't take the time to think about the changes that are made, what their impact is, and ask the author for clarifications/explanations (this is not a criticism of anyone's work, I have been the author of poorly tested PRs and the author of poor reviews, merging code without properly assessing it)
 - Team agreements for critical tasks (making a release, resolving security incidents, investigating an urgent bug in prod): how to we work as a team to ensure that the burden of critical tasks is shared and that those critical tasks are resolved in a timely manner
 - Technical debt (TD): can developers have a separate backlog for TD and work on TD in a time-boxed manner (e.g. 1 day of development per sprint)
 - Bugs triaging & backlog: we should discuss this before we start getting bug reports. Having probability and impact reported in a bug issue template would be good.
 - DoR & team velocity: to meet our DoR team agreements our velocity must be reduced. We have experienced many instances of the scope of PRs ballooning.

**Comment by matthieu-foucault** [comment]
In progress - first session happened. Follow-up sessions will be booked
